src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -265,12 +295,106 @@ private String GenerateCodeFlowDescriptionForSubscription(
+            }
+        }
+
+        if (dependencyCategories.RemovedDependencies.Count > 0)
+        {
+            stringBuilder.AppendLine();
+            stringBuilder.AppendLine("**Removed Dependencies**");
+            foreach (DependencyUpdateSummary depUpdate in dependencyCategories.RemovedDependencies)
+            {
+                stringBuilder.AppendLine($"- **{depUpdate.DependencyName}**: removed version {depUpdate.FromVersion}");
+            }
+        }
+
+        if (dependencyCategories.UpdatedDependencies.Count > 0)
+        {
+            stringBuilder.AppendLine();
+            stringBuilder.AppendLine("**Updated Dependencies**");
+            foreach (DependencyUpdateSummary depUpdate in dependencyCategories.UpdatedDependencies)
+            {
+                string diffLink = GetLinkForDependencyItem(repoUri, depUpdate);
+                stringBuilder.AppendLine($"- **{depUpdate.DependencyName}**: [from {depUpdate.FromVersion} to {depUpdate.ToVersion}]({diffLink})");
+            }
+        }
+        return stringBuilder.ToString();
+    }
+
+    private DependencyCategories CreateDependencyCategories(
+        List<DependencyUpdateSummary> dependencyUpdateSummaries)
+    {
+        List<DependencyUpdateSummary> newDependencies = new();
+        List<DependencyUpdateSummary> removedDependencies = new();
+        List<DependencyUpdateSummary> updatedDependencies = new();
+
+        foreach (DependencyUpdateSummary depUpdate in dependencyUpdateSummaries)
+        {
+            if (string.IsNullOrEmpty(depUpdate.FromVersion))
+            {
+                newDependencies.Add(depUpdate);
+            }
+            else if (string.IsNullOrEmpty(depUpdate.ToVersion))
+            {
+                removedDependencies.Add(depUpdate);
+            }
+            else
+            {
+                updatedDependencies.Add(depUpdate);
+            }
+        }
+
+        return new DependencyCategories(newDependencies, removedDependencies, updatedDependencies);
+    }
+
+    private static string GetLinkForDependencyItem(string repoUri, DependencyUpdateSummary dependencyUpdateSummary)
+        {
+        if (!string.IsNullOrEmpty(dependencyUpdateSummary.FromCommitSha) && !string.IsNullOrEmpty(dependencyUpdateSummary.ToCommitSha))
+        {
+            return GetChangesURI(repoUri, dependencyUpdateSummary.FromCommitSha, dependencyUpdateSummary.ToCommitSha);
+        }
+        else if (!string.IsNullOrEmpty(dependencyUpdateSummary.ToCommitSha))
+        {
+            if (repoUri.Contains("github.com"))
+            {
+                return $"{repoUri}/commit/{dependencyUpdateSummary.ToCommitSha}";
+            }
+            else if (repoUri.Contains("dev.azure.com"))
+            {
+                return $"{repoUri}?_a=history&version=GC{dependencyUpdateSummary.ToCommitSha}";
+            }
+        }
```

Comment:
Not sure how to apply the rules we discussed but isn't the following more readable?

```suggestion
        if (!string.IsNullOrEmpty(dependencyUpdateSummary.FromCommitSha) && !string.IsNullOrEmpty(dependencyUpdateSummary.ToCommitSha))
        {
            return GetChangesURI(repoUri, dependencyUpdateSummary.FromCommitSha, dependencyUpdateSummary.ToCommitSha);
        }

        if (string.IsNullOrEmpty(dependencyUpdateSummary.ToCommitSha))
        {
            return null;
        }
        
        if (repoUri.Contains("github.com"))
        {
            return $"{repoUri}/commit/{dependencyUpdateSummary.ToCommitSha}";
        }
        
        if (repoUri.Contains("dev.azure.com"))
        {
            return $"{repoUri}?_a=history&version=GC{dependencyUpdateSummary.ToCommitSha}";
        }
        
        return null;
```